### PR TITLE
feat: add ci workflows for go-redis-momento client

### PR DESCRIPTION
**Pull request by pratik151192**
Added CI workflows for go-redis compatibility client

### CODE REVIEW

File: .github/workflows/build.yml
Comment (pgautier404) [inline .github/workflows/build.yml:1]: It looks like this is just checking out the repository and nothing else. Did you intend for something else to happen here? If not, can we just remove this?

Comment (pratik151192) [reply .github/workflows/build.yml:1]: good point

File: .github/workflows/manual_release.yml
Comment (pgautier404) [inline .github/workflows/manual_release.yml:1]: Looking at the [Go SDK's manual release action,](https://github.com/momentohq/client-sdk-go/blob/main/.github/workflows/manual_release.yml) it actually looks like it's not even using a release branch. It's just using the semantic release plugin to calculate/tag the new version and then using the `publish_golang` step to verify the new version in the Go package repository. I think you'll want to use that as a model for how to tag/release your package.

Comment (pgautier404) [reply .github/workflows/manual_release.yml:1]: The SDK release action also has a `test` job that you should go ahead and add here as well.

File: .github/workflows/on_pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        uses: momentohq/standards-and-practices/github-actions/shared-build@gh-actions-v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  readme:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.MOMENTO_MACHINE_USER_GITHUB_TOKEN }}
```
Comment (pgautier404) [inline .github/workflows/on_pull_request.yml]: This should be removed. It's just checking out the repo and then moving on.

File: .github/workflows/manual_release.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          set -e
+          set -x
+          export COMPATIBILITY_CLIENT_VERSION="${{needs.release.outputs.version}}"
+          if [ -z "$COMPATIBILITY_CLIENT_VERSION"]
+          then
+            echo "Unable to determine Compatibility Client version!  Exiting!"
+            exit 1
+          fi
+          echo "COMPATIBILITY_CLIENT_VERSION=${COMPATIBILITY_CLIENT_VERSION}"
+          GOPROXY=proxy.golang.org go list -m github.com/momentohq/client-sdk-go@v${COMPATIBILITY_CLIENT_VERSION}
```
Comment (pgautier404) [inline .github/workflows/manual_release.yml]:
```suggestion
          GOPROXY=proxy.golang.org go list -m github.com/momentohq/momento-go-redis-client@v${COMPATIBILITY_CLIENT_VERSION}
```

Comment (pratik151192) [reply .github/workflows/manual_release.yml]: uhhh i want your eyes! :D

**Review by pgautier404 — APPROVED** [review]
:shipit:

**Pull request merged**